### PR TITLE
Document events, stripe and add events to pricing and contact fragments

### LINKED ISSUE #485: [pricing] document pricing, events and stripe usage
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
We document each fragment, but we don't document the interconnection.

**What you expected to happen**:
Stripe could mention that price changes can happen via the events fragment and the pricing fragment.

**Anything else we need to know?**:
rough idea:
*  events: url -> event + develop/use events (link to pricing for an example of an emitter, link to stripe for an event user)
*  stripe: change price via events with pricing or url (link to pricing and events)
*  pricing: emit price events (link to events)

### LINKED ISSUE #539: [docs] Document and explain base64 encoded event urls

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:
- [x] Add documentation for events
- [x] Add documentation for stripe
- [x] Add documentation for pricing

**Which issue this PR fixes**:
fixes #485 
fixes #539 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by mpourismaiel** [comment]
@stp-ip This PR seems finished.

File: exampleSite/content/fragments/pricing/docs.md [hunk 1]
```diff
@@ -33,6 +37,13 @@ Title of the button on the column.
 
 URL of the button on the column.
 
+##### Usage with events
+
+You can make use of the Events api through this variable.
+
+- Not setting the `button_url` variable will publish an event with `title`, `subtitle`, `price` and `currency` variables.
+- Setting it to an event url such as: `/dev/events/payment?event=pricing:change&product=Starting plan&price=$9.99/mo&currency=usd` will redirect the page and publish a custom event.
```
Comment (stp-ip) [inline exampleSite/content/fragments/pricing/docs.md]:
```suggestion
- Setting it to an event url such as: `/fragments/stripe/?event=pricing:change&product=Starting plan&price=$9.99/mo&currency=usd` will redirect the page and publish a custom event.
```

File: exampleSite/content/fragments/stripe/docs.md [hunk 2]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
   enable = true
 +++
 
-# UNDER CONSTRUCTION
+Stripe fragment creates a form with a Stripe element inside it and is used to create payments for the users.
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]:
```suggestion
Stripe fragment creates a form with a Stripe element inside, which is used to create payments for the users.
```

File: exampleSite/content/fragments/stripe/docs.md [hunk 3]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+
+It requires a backend for creating the customer and charge for Stripe. An example can be found at [here](https://github.com/okkur/syna-stripe-gcf) which can be deployed on Google Cloud Functions and works out of the box.
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]:
```suggestion
It requires a backend for creating the customer and charge for Stripe. An example can be found at [here](https://syna.okkur.org/stripe-gcf) which can be deployed on Google Cloud Functions and works out of the box.
```

File: exampleSite/content/fragments/stripe/docs.md [hunk 4]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+
+The use of a backend is also useful for saving other information from the user such as the product they selected (in case you are providing more than one products) or other metadata.
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]:
```suggestion
The use of a backend is also useful for saving other information/metadata from the user such as the product they selected.
```

File: exampleSite/content/fragments/stripe/docs.md [hunk 5]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+
+###### product
+*type: string*
+
+If provided, this variable will override the preconfigured `product` variable and will be displayed on top of the form.
+
+###### description
+*type: string*
+
+Payment metadata. This varialbe is not displayed to the user but will be sent to the server with the form data.
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]:
```suggestion
Payment metadata. This variable is not displayed to the user but will be sent to the server with the other form data.
```

File: exampleSite/content/fragments/stripe/docs.md [hunk 6]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+
+###### price
+*type: string*
+
+The price that will be charged to the user's account.
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]: Probably good to give multiple example values here.

File: exampleSite/content/fragments/stripe/docs.md [hunk 7]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+
+###### currency
+*type: string*
+
+The currency in which the user will be charged.
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]: Maybe add a small note how the price value and it's corresponding currency such as "$" is only user facing and the currency variable is essential for determining the charge. What do you think?

File: exampleSite/content/fragments/stripe/docs.md [hunk 8]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+
+### Variables
+
+#### post_url
+*type: string*
+
+Form data will be sent to this url. The url should point to an api which can handle Stripe's charge customer api. An [example](https://github.com/okkur/syna-stripe-gcf) is provided for more information.
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]:
```suggestion
Form data will be sent to this url. The url should point to an api which can handle Stripe's charge customer api. An [example](https://syna.okkur.org/stripe-gcf) is provided for more information.
```

File: exampleSite/content/fragments/stripe/docs.md [hunk 9]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+#### product
+*type: string*
+
+Your product's name. This variable is optional.
+
+#### email
+*type: object*
+
+##### email.label
+*type: string*
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]: What does the email.label do and why do we need the split from email and email.label?

File: exampleSite/content/fragments/stripe/docs.md [hunk 10]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+
+#### prices
+*type: array of objects*
+
+This variable is optional. If the array contains a single item, the form's design will change to indicate that the price cannot be changed. If no values are provided, the form will fallback to `user_input`.
+
+##### prices.text
+*type: string*
+
+This variable is shown on the price button. On form submission it will be parsed into the format that Stripe expects.
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]: Maybe add a small note how the price value and it's corresponding currency such as "$" is only user facing and the currency variable is essential for determining the charge. What do you think?

File: exampleSite/content/fragments/stripe/docs.md [hunk 11]
```diff
@@ -8,6 +8,92 @@ title = "Documentation"
...
+This variable will create a field and will display either the field itself or a button that will switch the form to editable mode. The field is displayed based on whether the `prices` field contains any items or not.
+
+##### user_input.default
+*type: string*
+
+Default value in the editable price field.
+
+##### user_input.currencies
+*type: array of strings*
+
```
Comment (stp-ip) [inline exampleSite/content/fragments/stripe/docs.md]: Add example of how a currency is specified aka 3 char abbreviation.
Add link to list of currencies supported from stripe and their specific abbreviations.

File: docs/fragments/contact.md [hunk 12]
```diff
@@ -18,7 +18,7 @@ Subtitle of the form. This is displayed in the section shows the form.
 ### post_url *string* *Either **post_url** or **email** should be provided*
 If set, submitted data would be sent to this url.
 ### email *string* *Either **post_url** or **email** should be provided*
-If set and there is no **post_url** provided, then the form data would be sent to `https://formspree.io/[email]`. This is the default behavior and can be overriden using the `post_url` variable.
+If set and there is no **post_url** provided, then the form data would be sent to `https://formspree.io/[emsail]`. This is the default behavior and can be overriden using the `post_url` variable.
```
Comment (stp-ip) [inline docs/fragments/contact.md]: Added typo.

**Comment by stp-ip** [comment]
Merge from master needed as this should remove resources as this is now in gitignore.
Beside the small comment above. All seems ready to merge. Waiting for the conflict resolution and the typo fix. Then merge \o/

File: docs/fragments/contact.md (hunk 12 above)
Comment (mpourismaiel) [reply docs/fragments/contact.md]: This page is heavily damaged. The variables and their types and descriptions are in the same line. Also I think this page will be removed when new docs page goes out along with v0.15. I fixed the typo, should I fix the problems as well for now or should we wait and delete it later? BTW I'm not sure why I changed email in here but that was an accident.

Comment (stp-ip) [reply docs/fragments/contact.md]: Let's fix the specific docs in another PR.

Typo fix aka no change seems like a good thing to do here.

**Pull request merged**